ui/app/src/components/AddProjectDialog/AddProjectDialog.tsx

```diff
@@ -0,0 +1,125 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ChangeEvent, Dispatch, DispatchWithoutAction, useCallback, useState } from 'react';
+import { Dialog, DialogTitle, DialogContent, DialogActions, Button, TextField, IconButton, Theme } from '@mui/material';
+import CloseIcon from 'mdi-material-ui/Close';
+import { ProjectModel, useAddProjectMutation } from '../../model/project-client';
+import { useSnackbar } from '../../context/SnackbarProvider';
+
+/**
+ * Render the CSS of the dialog's close button, according to the given material theme.
+ * @param theme material theme
+ */
+const dialogCloseIconButtonStyle = function (theme: Theme) {
+  return { position: 'absolute', top: theme.spacing(0.5), right: theme.spacing(0.5) };
+};
+
+export interface AddProjectDialogProps {
+  open: boolean;
+  onClose: DispatchWithoutAction;
+  onSuccess?: Dispatch<ProjectModel>;
+}
+
+/**
+ * Dialog used to create a project.
+ * @param props.open Define if the dialog should be opened or not.
+ * @param props.closeDialog Provides the function to close itself.
+ * @param props.onConfirm Action to perform when user confirmed.
+ * @constructor
+ */
+const AddProjectDialog = (props: AddProjectDialogProps) => {
+  const { open, onClose, onSuccess } = props;
+  const [name, setName] = useState<string>('');
+  const [error, setError] = useState<string>('');
+
+  // Called every time the user type in the form
+  const handleNameFormChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      setName(e.target.value);
+      if (!e.target.value) {
+        setError('Required');
+      } else if (!/^[a-zA-Z0-9_.:-]+$/.exec(e.target.value)) {
+        setError('Allowed special characters are _ . : -');
+      } else {
+        setError('');
+      }
+      // TODO: Verify the non-existence of the project, using a debounce (300ms?) to not call it too many
+    },
+    [setName, setError]
+  );
+
+  // Reinitialize form for next time the dialog is opened
+  const handleNameFormReset = useCallback(() => {
```

Comment:
Just a suggestion but i think if you create another component for dialog content, you don't have to worry about resetting the states. 

```
return (
  <Dialog open={open}>
     { open && <AddProjectDialogContent /> }  
  </Dialog>
);
```

Reply:
Interesting, let me try and think about it but I'm not sure it's more readable. The "formReset" is more self explanatory. Last time I played around with some mount* unmount* property I don't remember to force the unmount when closed but without success. I will try again.

Reply:
I'm going to go ahead and merge this PR since I want to test with #939 and potentially do a release. Maybe this can be looked into as a follow up after #948 is merged.